src/KafkaClient/Protocol/KafkaEncoder.cs

```diff
@@ -243,7 +243,7 @@ private static IResponse MetadataResponse(IRequestContext context, ArraySegment<
                 }
 
                 int? controllerId = null;
-                if (context.ApiVersion >= 1) {
+                if (context.ApiVersion >= 2) {
                     controllerId = reader.ReadInt32();
```

Comment:
controller_id is part of version 1 as well:

Metadata Response (Version: 1) => [brokers] controller_id [topic_metadata] 
  brokers => node_id host port rack 
    node_id => INT32
    host => STRING
    port => INT32
    rack => NULLABLE_STRING
  controller_id => INT32
...